Install/Compile issues with node v0.12.0

Trying to install this on node v0.12.0 gives a bunch of errors. I'm guessing nan 0.8 isn't compatible with node v0.12

```
> mac-key-press@0.1.0 install /Users/chinmay/node/lib/node_modules/blueslider/node_modules/mac-key-press
> node-gyp rebuild

child_process: customFds option is deprecated, use stdio instead.
  CXX(target) Release/obj.target/addon/addon.o
In file included from ../addon.cc:3:
../node_modules/nan/nan.h:339:13: error: no member named 'New' in 'v8::String'
    return  _NAN_ERROR(v8::Exception::Error, errmsg);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:319:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:343:5: error: no member named 'ThrowException' in namespace
      'v8'; did you mean 'MPThrowException'?
    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:324:7: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
      ^~~~
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Multiprocessing.h:1556:1: note:
      'MPThrowException' declared here
MPThrowException(
^
In file included from ../addon.cc:3:
../node_modules/nan/nan.h:343:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:324:26: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
                         ^~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:319:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:348:9: error: no type named 'ThrowException' in namespace 'v8'
    v8::ThrowException(error);
    ~~~~^
../node_modules/nan/nan.h:355:65: error: no member named 'New' in 'v8::String'
    v8::Local<v8::Value> err = v8::Exception::Error(v8::String::New(msg));
                                                    ~~~~~~~~~~~~^
../node_modules/nan/nan.h:357:26: error: no member named 'New' in 'v8::String'
    obj->Set(v8::String::New("code"), v8::Int32::New(errorNumber));
             ~~~~~~~~~~~~^
../node_modules/nan/nan.h:369:12: error: no member named 'New' in 'v8::String'
    return _NAN_ERROR(v8::Exception::TypeError, errmsg);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:319:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:373:5: error: no member named 'ThrowException' in namespace
      'v8'; did you mean 'MPThrowException'?
    _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:324:7: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
      ^~~~
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Multiprocessing.h:1556:1: note:
      'MPThrowException' declared here
MPThrowException(
^
In file included from ../addon.cc:3:
../node_modules/nan/nan.h:373:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:324:26: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
                         ^~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:319:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:377:12: error: no member named 'New' in 'v8::String'
    return _NAN_ERROR(v8::Exception::RangeError, errmsg);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:319:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:381:5: error: no member named 'ThrowException' in namespace
      'v8'; did you mean 'MPThrowException'?
    _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:324:7: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
      ^~~~
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Multiprocessing.h:1556:1: note:
      'MPThrowException' declared here
MPThrowException(
^
In file included from ../addon.cc:3:
../node_modules/nan/nan.h:381:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:324:26: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
                         ^~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:319:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:727:49: error: too few arguments to function call, single
      argument 'isolate' was not specified
    v8::Local<v8::Object> obj = v8::Object::New();
                                ~~~~~~~~~~~~~~~ ^
/Users/chinmay/.node-gyp/0.12.0/deps/v8/include/v8.h:2388:3: note: 'New' declared here
  static Local<Object> New(Isolate* isolate);
  ^
In file included from ../addon.cc:3:
../node_modules/nan/nan.h:733:49: error: too few arguments to function call, single
      argument 'isolate' was not specified
    v8::Local<v8::Object> obj = v8::Object::New();
                                ~~~~~~~~~~~~~~~ ^
/Users/chinmay/.node-gyp/0.12.0/deps/v8/include/v8.h:2388:3: note: 'New' declared here
  static Local<Object> New(Isolate* isolate);
  ^
In file included from ../addon.cc:3:
../node_modules/nan/nan.h:740:12: error: no member named 'Dispose' in
      'v8::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
    handle.Dispose();
    ~~~~~~ ^
../node_modules/nan/nan.h:741:12: error: no member named 'Clear' in
      'v8::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
    handle.Clear();
    ~~~~~~ ^
../node_modules/nan/nan.h:746:39: error: no member named 'NewSymbol' in 'v8::String'
    NanPersistentToLocal(handle)->Set(NanSymbol("callback"), fn);
                                      ^~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:181:38: note: expanded from macro 'NanSymbol'
#define NanSymbol(value) v8::String::NewSymbol(value)
                         ~~~~~~~~~~~~^
../node_modules/nan/nan.h:750:46: error: no member named 'NewSymbol' in 'v8::String'
    return NanPersistentToLocal(handle)->Get(NanSymbol("callback"))
                                             ^~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:181:38: note: expanded from macro 'NanSymbol'
#define NanSymbol(value) v8::String::NewSymbol(value)
                         ~~~~~~~~~~~~^
../node_modules/nan/nan.h:751:25: error: expected '(' for function-style cast or type
      construction
        .As<v8::Function>();
            ~~~~~~~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/addon/addon.o] Error 1
```

--- a/addon.cc
+++ b/addon.cc
@@ -8,12 +8,12 @@ NAN_METHOD(Press) {
   NanScope();
 
   if (args.Length() < 1) {
-    ThrowException(Exception::TypeError(String::New("Wrong number of arguments")));
+    NanThrowTypeError("Wrong number of arguments");
     NanReturnUndefined();
   }
 
   if (!args[0]->IsNumber()) {
-    ThrowException(Exception::TypeError(String::New("Wrong arguments")));
+    NanThrowTypeError("Wrong arguments");
     NanReturnUndefined();
   }
 
@@ -32,12 +32,12 @@ NAN_METHOD(Move) {
   NanScope();
 
   if (args.Length() < 2) {
-    ThrowException(Exception::TypeError(String::New("Wrong number of arguments")));
+    NanThrowTypeError("Wrong number of arguments");
     NanReturnUndefined();
   }
 
   if (!args[0]->IsNumber() || !args[1]->IsNumber()) {
-    ThrowException(Exception::TypeError(String::New("Wrong arguments")));
+    NanThrowTypeError("Wrong arguments");
     NanReturnUndefined();
   }
 
@@ -46,7 +46,7 @@ NAN_METHOD(Move) {
   size_t yMax = CGDisplayPixelsHigh(kCGDirectMainDisplay);
 
   if (args[0]->NumberValue() > xMax || args[0]->NumberValue() < 0 || args[1]->NumberValue() > yMax || args[1]->NumberValue() < 0){
-    ThrowException(Exception::TypeError(String::New("Invalid cursor position")));
+    NanThrowTypeError("Invalid cursor position");
     NanReturnUndefined();
   }
 
@@ -55,7 +55,7 @@ NAN_METHOD(Move) {
   CFRelease(move);
 
   if(move == NULL) {
-    ThrowException(Exception::TypeError(String::New("Error creating the event")));
+    NanThrowTypeError("Error creating the event");
     NanReturnUndefined();
   }
 
@@ -69,21 +69,21 @@ NAN_METHOD(GetPos) {
   CGPoint cursor = CGEventGetLocation(event);
   CFRelease(event);
 
-  Local<Object> pos = Object::New();
+  Local<Object> pos = NanNew<Object>();
 
-  pos->Set(String::NewSymbol("x"), Number::New(cursor.x));
-  pos->Set(String::NewSymbol("y"), Number::New(cursor.y));
+  pos->Set(NanNew<String>("x"), NanNew<Number>(cursor.x));
+  pos->Set(NanNew<String>("y"), NanNew<Number>(cursor.y));
 
   NanReturnValue(pos);
 }
 
 void Init(Handle<Object> exports) {
-  exports->Set(String::NewSymbol("press"),
-      FunctionTemplate::New(Press)->GetFunction());
-  exports->Set(String::NewSymbol("move"),
-      FunctionTemplate::New(Move)->GetFunction());
-  exports->Set(String::NewSymbol("getPos"),
-      FunctionTemplate::New(GetPos)->GetFunction());
+  exports->Set(NanNew<String>("press"),
+      NanNew<FunctionTemplate>(Press)->GetFunction());
+  exports->Set(NanNew<String>("move"),
+      NanNew<FunctionTemplate>(Move)->GetFunction());
+  exports->Set(NanNew<String>("getPos"),
+     NanNew<FunctionTemplate>(GetPos)->GetFunction());
 }
 
 NODE_MODULE(addon, Init)

--- a/package.json
+++ b/package.json
@@ -10,6 +10,6 @@
   "author": "Matteo Collina <hello@matteocollina.com>",
   "license": "MIT",
   "dependencies": {
-    "nan": "~0.8.0"
+    "nan": "~1.7.0"
   }
 }

